wolfcrypt/src/port/maxim/max3266x.c

```diff
@@ -0,0 +1,921 @@
+/* max3266x.c
+ *
+ * Copyright (C) 2006-2023 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#warning "Inside max3266x.c"
+#if defined(WOLFSSL_MAX32665) ||  defined(WOLFSSL_MAX32666)
+#warning "Inside max3266x.c define"
+
+
+#ifdef NO_INLINE
+    #include <wolfssl/wolfcrypt/misc.h>
+#else
+    #define WOLFSSL_MISC_INCLUDED
+    #include <wolfcrypt/src/misc.c>
+#endif
+
+#include <stdint.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/wolfcrypt/logging.h>
+#include <wolfcrypt/src/misc.c>
+#include <wolfssl/wolfcrypt/port/maxim/max3266x.h>
+#include <wolfssl/wolfcrypt/types.h>
+
+#if defined(USE_FAST_MATH) && defined(WOLFSSL_SP_MATH_ALL)
```

Comment:
These rules are already handled in settings.h. Please just report error when `USE_FAST_MATH` is set. Please remove the `#warning`.